### PR TITLE
Add possibility to change the OOTB parameters

### PR DESCRIPTION
**Pull request by arnovanliere**
## Why
I wanted a method to set the default values for OOTB params which wasn't possible, because if you tried to register a verbose parameter again, goyek would panic because it was already registered.

## What
I added a new method to create a flow (`NewFlow`) which accepts an `Options`-struct in which the defaults can be set.

## Checklist
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated.
- [x] The code changes follow [Effective Go](https://golang.org/doc/effective_go).

### CODE REVIEW

**Comment by pellared** [comment]
@arnovanliere I see your problem. I have another idea of how we can approach it.

Would it be acceptable if you could change the defaults (and design) in the following way?

```go
flow := &goyek.Flow{}
flow.RegisterWorkDirParam(StringParam{
	Name:    "wd",
	Usage:   "Working directory: set the working directory.",
	Default: "./build",
})
flow.RegisterVerboseParam(BoolParam{
	Name:  "verbose",
	Usage: "Verbose: log all tasks as they are run.",
	Default: true,
})
```

These functions could set the `workDir` and `verbose` fields. It would allow changing other fields such as `Name` and `Usage`. Also, it would not require adding a `NewFlow` function.

Please let me know what you think.

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ This release contains multiple **breaking changes** in the Go API. It is suppose
 
 - `Tf.Cmd` also sets `Stdin` to `os.Stdin`.
 - Add `Flow.Tasks` and `Flow.Params` to allow introspection of the registered tasks and parameters.
+- Add `NewFlow` to create a new flow with custom defaults for default parameters.
```
Comment (pellared) [inline CHANGELOG.md:18]: This should be in `Unreleased` section 😉

**Comment by arnovanliere** [comment]
@pellared probably a better solution. I'll update my mergerequest and get back to you.

File: CHANGELOG.md (hunk 1 above)
Comment (arnovanliere) [reply CHANGELOG.md]: I've moved it.

**Comment by arnovanliere** [comment]
@pellared I've implemented your proposal. Could you please take another look?

**Comment by pellared** [comment]
LGTM, just fix the `markdownlint` errors 😉

**Comment by arnovanliere** [comment]
Just fixed them. Added myself to the license as well. Thanks!

**Review by pellared — COMMENTED** [review]
Can you add tests for both `RegisterVerboseParam` and `RegisterWorkDirParam`, please? You can use `Test_wd_param` as a reference test.

PS. Make sure to pull the changes as I merged `main` into your branch.

**Comment by arnovanliere** [comment]
I've added a few test cases. I don't know if this is sufficient?

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
@arnovanliere 

> The base branch requires all commits to be signed. [Learn more about signing commits.](https://docs.github.com/articles/about-gpg/)

This is the last step 😄 It is crapy that GitHub does not mention it sooner...

BTW would it help you if you had some `CONTRINUTING.md` with more details describing how to contribute?

**Comment by arnovanliere** [comment]
Thanks for removing the requirement as I've never worked with signed commits.

A `CONTRINUTING.md` file might help I think as the checklist when opening a pull-request is rather short.

**Comment by pellared** [comment]
@arnovanliere It was already there https://github.com/goyek/goyek/blob/main/docs/contributing.md. Have you seen it? Is anything missing there?

Moreover, I have just released the new version. I would be also grateful if you let me know if you are happy with `goyek` or is there any better alternative or you think something could be changed.

**Comment by arnovanliere** [comment]
I guess I missed it. I expected in the root of the repository. I would suggest adding a complete checklist in this file so people know what they have to do to get a pull request approved (like adding tests etc). It would be nice if there was a section in the `README.md` which points to the contributing file so people know where to find it.

I'm very happy with `goyek`. I work an a big API repository and at first we had like 20 to 30 Bash scripts which were unreadable, but with `goyek`, we nicely condensed it and it is easier to parallelize tasks as well.

Thanks for the new release!

**Pull request merged**